rgw/v2/tests/s3_swift/test_swift_basic_ops.py

```diff
@@ -64,51 +80,79 @@ def test_exec(config):
-            log.info('download object name: %s' % swift_object_download_fname)
-            swift_object_download_path = os.path.join(TEST_DATA_PATH, swift_object_download_fname)
-            log.info('download object path: %s' % swift_object_download_path)
-            swift_object_downloaded = rgw.get_object(container_name, swift_object_name)
-            with open(swift_object_download_path, 'w') as fp:
-                fp.write(str(swift_object_downloaded[1]))
-            # modify and re-upload
-            log.info('appending new message to test_data')
-            message_to_append = 'adding new msg after download'
-            fp = open(swift_object_download_path, 'a+')
-            fp.write(message_to_append)
-            fp.close()
-            with open(swift_object_download_path, 'r') as fp:
-                rgw.put_object(container_name, swift_object_name,
-                               contents=fp.read(),
-                               content_type='text/plain')
-            # delete object
-            log.info('deleting swift object')
-            rgw.delete_object(container_name, swift_object_name)
-        # delete container
-        log.info('deleting swift container')
-        rgw.delete_container(container_name)
+        if config.test_ops.get('enable_version', True):
+            container_name_old = utils.gen_bucket_name_from_userid(user_info['user_id'], rand_no=str(cc) + 'old')
+            log.info(container_name_old)
+            container = swiftlib.resource_op({'obj': rgw,
+                                              'resource': 'put_container',
+                                              'kwargs': dict(container=container_name_old)})
+            container_name = utils.gen_bucket_name_from_userid(user_info['user_id'], rand_no=str(cc) + 'new')
+            log.info(container_name)
+            container = swiftlib.resource_op({'obj': rgw,
+                                              'resource': 'put_container',
+                                              'args': [container_name, {'X-Versions-Location': container_name_old}]})
+            if container is False:
+                raise TestExecError("Resource execution failed: container creation failed")
+            for version_count in range(config.version_count):
+                for oc, size in list(config.mapped_sizes.items()):
+                    swift_object_name = fill_container(rgw, container_name, user_names[0], oc, cc, size)
+            current_count = 'radosgw-admin bucket stats --uid={uid} --tenant={tenant} --bucket=\'{bucket}\' ' \
+                            '| grep num_objects'.format(uid=user_names[0], tenant=tenant, bucket=container_name)
+            num_obj_current = utils.exec_shell_cmd(current_count)
+            num_obj_current = num_obj_current.replace(" ", "")
+            num_obj_current = num_obj_current.replace("\n", "")
+            li = list(num_obj_current.split(":"))
+            num_obj_current = int(li[1])
```

Comment:
you can use strip to remove the `"\n"` and also clean up the variable usage here, there is definitely scope for better implementation here.